src/nucleus/jdg.ml

```diff
@@ -383,8 +387,8 @@ let match_arguments sgn (premises : Rule.premise list) (arguments : argument lis
        (* The arguments must _not_ be reversed because we refer to them by meta-variable
           de Bruijn indices, and therefore the last argument must have index 0. *)
        args_out
-    | [], _::_ -> failwith "too many arguments"
-    | _::_, [] -> failwith "too few arguments"
+    | [], _::_ -> TT.error TT.TooManyArguments
+    | _::_, [] -> TT.error TT.TooFewArguments
```

Comment:
These should be assert false / failwith, because src/parsing/desugar.ml:772 already checks arities of rules.